openmdao/core/problem.py

```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
         """
         Initialize attributes.
         """
+        global _problem_counter
+
         self.cite = CITATION
-        self._name = name
+
+        # Code to give non-empty names to Problems so that they can be
+        # referenced from command line tools (e.g. check) that accept a Problem argument
+        _problem_counter += 1
+        if name:  # if name hasn't been used yet, use it. Otherwise, error
+            if name not in _problem_names:
+                self._name = name
+            else:
+                raise ValueError(f"The problem name '{name}' already exists")
+        else:  # No name given: look for a name, of the form, 'problemN', that hasn't been used yet
+            _name = f"problem{_problem_counter}"
```

Comment:
I think I understand the logic here now. If we are the second problem, but "problem2" is taken, we just start with "problem2.1", and keep working until we find one that isn't taken.